### PR TITLE
PSR-2 PHPStorm and PHP-CS-Fixer problems

### PR DESCRIPTION
**Pull request by gvollbach**
It seems there where three options missing in the php-cs-fixer config, which are now included:

1. [Align operators](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16|fixer:binary_operator_spaces)
2. [Order imports](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16|fixer:ordered_imports)
3. [No extra consecutive blank lines](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16|fixer:no_extra_blank_lines)

If there is still an option missing, please feel free to report this.

@ILIAS-eLearning/technical-board 
Please review and decide which of these three options we should activate or deactivate. The manual to the options are linked above, examples are on the specific pages. 
All three options are now active with this config and are now the same as the default PHPStorm codestyle.

### CODE REVIEW

**Comment by klees** [comment]
Hi @gvollbach,

thanks for looking into this.

Your changes are only in the CS-Fixer config to make it match the PHPStorm template, but it could as well be required to move the PHPStorm-template into the other direction to match our [coding style](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/coding-style.md). So I had a look into the document and tried to find answers. They are:

* [x] Please remove [align operators](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16%7Cfixer:binary_operator_spaces) and change it to `single space`, in the CS-Fixer config as well as in the PHPStorm template. Our [coding style (regarding operators)](https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/coding-style.md#operators) says "Operators MUST have a space before and after the operator.".
* [x] Please remove [ordered imports](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16|fixer:ordered_imports), in the CS-Fixer config as well as in the PHPStorm template. I could not find any rule that requires the sortation of imports.
* [x] Please remove [no extra blank lines](https://mlocati.github.io/php-cs-fixer-configurator/#version:2.16%7Cfixer:no_extra_blank_lines), in the CS Fixer config as well as in the PHPStorm template. [PSR-2](https://www.php-fig.org/psr/psr-2/), which is the base line for our coding style, explicitly allows blank lines: "Blank lines MAY be added to improve readability and to indicate related blocks of code.".

Personally, I would sympathize with no "extra blank lines" (i.e. no more than *one*) and also "ordered imports" (and disagree to the alignment), but this is only about consistently implementing the style we agreed on, not defining what the style is, so no personal taste involved here =)

Best regards!

**Review by klees — CHANGES_REQUESTED** [review]
:point_up_2:

File: CI/PHP-CS-Fixer/example/example_cleaned.php [hunk 1]
```diff
@@ -27,16 +28,19 @@ public function sampleMethod($a, $b = null)
 
 /** thx to https://mlocati.github.io/php-cs-fixer-configurator for the examples **/
 // cast_spaces
-$a = 0; $b = 0; $c = 0; $d = 0; $e = 0 ; $f = 0;
-$bar = (string) $a;
-$foo = (int) $b;
+$a    = 0; $b    = 0; $c    = 0; $d   =0; $e   = 0 ; $f    =0;
+$bar  = (string) $a;
+$foo  =  (int) $b;
```
Comment (alex40724) [inline CI/PHP-CS-Fixer/example/example_cleaned.php:33]: @gvollbach 
I am confused, the example now alignes the operators, but I thought we do not want that.

**Review by alex40724 — COMMENTED** [review]
Thanks for taking care of this. I can confirm that operator alignment is an issue. However the changes in the example seem to add alignments now. Or am I reading the diff wrongly?

Edit: Ok I just walked through the commits, the example is just missing an update, right?

File: CI/PHP-CS-Fixer/example/example_cleaned.php (hunk 1 above)
Comment (gvollbach) [reply CI/PHP-CS-Fixer/example/example_cleaned.php:33]: The option is already removed from the config, but the example files are not cleaned, I will take care of this if all options are fixed.

**Comment by alex40724** [comment]
@mjansenDatabay @gvollbach There are some todos left, right? Could you give a status update please?

**Comment by gvollbach** [comment]
@alex40724 
I will take a look at this again next month.

**Comment by alex40724** [comment]
@gvollbach Thanks!

**Comment by gvollbach** [comment]
I tried so far to get this to code style checkers to behave similar, but I think the PHPStorm code checker will still behave a little different than the php-cs-fixer. I'm also not entirely sure that we will get both tools to behave entirely similar.

So if you still find corner cases where the tools differ, please post an example or add the example to the `example_raw.php` file, than I can take another look at this.

File: CI/PHP-CS-Fixer/code-format.php_cs [hunk 2]
```diff
@@ -26,5 +26,8 @@ return PhpCsFixer\Config::create()
         'unary_operator_spaces' => true,
         'function_typehint_space' => true,
         'return_type_declaration' => ['space_before' => 'one'],
+        'binary_operator_spaces' => [
+         'operators' => ['=' => 'align_single_space']
```
Comment (klees) [inline CI/PHP-CS-Fixer/code-format.php_cs]:
```suggestion
         'operators' => ['=' => 'single_space']
```

**Review by klees — CHANGES_REQUESTED** [review]
Hi @gvollbach,

I think the operators will still be aligned with the current change, which we imo did not agree upon. I added the according change below. I'm not sure what this means for the PHP-Storm template.

Thanks for your offer to have closer looks later on, though, I also agree that there will be edge cases left.

After this is finished I will reapply the coding style to all active branches, then we can watch the violations on the dashboard and changes in PRs again...

Best regards!

File: CI/PHP-CS-Fixer/code-format.php_cs (hunk 2 above)
Comment (gvollbach) [reply CI/PHP-CS-Fixer/code-format.php_cs]: @klees thx for the review, the config line is changed.

**Pull request merged**